### ISSUE TITLE
[Windows] Automaton isn't supported

### ISSUE THREAD

**Issue by gpotter2** [secdev/scapy#451]
**Edit: Fixed**

Here is the state of my investigation on the automaton tests (under windows with `wait=False`)

- The code bugs on windows because of this line:
https://github.com/secdev/scapy/blob/master/scapy/automaton.py#L625

- Why ? 
Using debug logs, i've found that [this](https://github.com/secdev/scapy/blob/master/scapy/automaton.py#L565) returned an error containing:

> error(10038, "An operation was tried on something that is not a socket.")  [[1]*](https://support.microsoft.com/en-us/kb/819124)

## Explanation

From https://docs.python.org/3/library/select.html#select.select :
> File objects on Windows are not acceptable, but sockets are. On Windows, the underlying select() function is provided by the WinSock library, and does not handle file descriptors that don’t originate from WinSock.

And in the code, `select` try to read on `fds = [self.cmdin]`. This does not originate from WinSock at all... It originates from https://github.com/secdev/scapy/blob/master/scapy/automaton.py#L21 custom object.

See: [this stack over flow question](http://stackoverflow.com/questions/35889267/an-operation-was-attempted-on-something-that-is-not-a-socket-tried-fixing-a-lot/35889952#35889952) for people having the same problem...

## To sum up:

on windows, `select` either requires a winsock object created using the [socket](https://docs.python.org/2/library/socket.html) api, either is missing something... The object generated with `os.pipe` doesn't look to be a socket on windows , which fails the code...

## Now, how do we fix it ?

I don't really know how to fix this...

Some people have found a solution for their problems, this one looks somehow to ours: http://stackoverflow.com/questions/18786128/select-and-pipes-trouble-in-python

Or we might have to create our own sort of pipe system, or avoid using select...

**Comment by gpotter2** [comment]
Finally fixed in https://github.com/secdev/scapy/pull/496
